### PR TITLE
Update generated tests; add shell script to avoid people working out how

### PR DESCRIPTION
**Pull request by gsnedders**
At the moment, most of our generated tests are badly documented as to how to generate them. One solution is to just provide a Makefile that does all the heavy lifting (and ultimately, hopefully, have a Makefile at the root which just regenerates all tests with a generation step, and ideally we can then run that on Travis CI to ensure they all stay up to date). This does that, and regenerates all the tests. And fixes the breakage caused by people not updating the source files properly.

### CODE REVIEW

File: html/semantics/embedded-content/the-canvas-element/size.attributes.setAttribute.whitespace.html
Comment (Ms2ger) [inline html/semantics/embedded-content/the-canvas-element/size.attributes.setAttribute.whitespace.html]: What happened here?

**Comment by Ms2ger** [comment]
Is there any reason to use make rather than a plain shell script?

File: html/semantics/embedded-content/the-canvas-element/size.attributes.setAttribute.whitespace.html
Comment (gsnedders) [inline html/semantics/embedded-content/the-canvas-element/size.attributes.setAttribute.whitespace.html]: https://github.com/w3c/web-platform-tests/commit/092f472231935e7478c59f8bdf149b991858b722

**Comment by gsnedders** [comment]
@Ms2ger not really, depends how smart we want to be about it

**Comment by sideshowbarker** [comment]
@gsnedders ping

**Comment by gsnedders** [comment]
Hmm, I think we might not have the parser tests being run properly. We have Chrome's stability test timing out, and quite a lot of error messages from Firefox…

**Comment by gsnedders** [comment]
OK, timeouts in parser tests:

- [ ] html/syntax/parsing/html5lib_innerHTML_foreign-fragment.html
- [ ] html/syntax/parsing/html5lib_innerHTML_tests_innerHTML_1.html
- [ ] html/syntax/parsing/html5lib_scriptdata01.html
- [ ] html/syntax/parsing/html5lib_template.html
- [ ] html/syntax/parsing/html5lib_tests1.html
- [ ] html/syntax/parsing/html5lib_tests19.html
- [ ] html/syntax/parsing/html5lib_tests20.html
- [ ] html/syntax/parsing/html5lib_tests21.html

**Comment by gsnedders** [comment]
Those were all caused by duplicates in the test object, now fixed.

**Comment by gsnedders** [comment]
@jgraham, @Ms2ger, @sideshowbarker r?

**Review by sideshowbarker — APPROVED** [review]
We should at some point also update the docs to include a notice about running the top-level `update-built-tests.sh` script, but I don’t think we should block merging this on waiting for that to be added.

**Comment by sideshowbarker** [comment]
LGTM

I tested this in my (MacOS) environment (including making some changes to test sources and ensuring the tests get regenerated) and it works as expected.

Code-wise I think @jgraham should review this (especially the python changes).

**Comment by gsnedders** [comment]
@jgraham can you please review this per the above comment?

**Comment by jdm** [comment]
We could add a travis job that runs the shell script and ensures that `git diff` is empty.

**Comment by gsnedders** [comment]
@jdm Bah, that won't work 100% of the time as long as we rely on html5lib-python's master branch, actually. :\

Now, the relevant question is can we live with this if we assume that failures down to this will be rare, and we should be able to update it pretty quickly given it's just rebuilding tests and hence no review needed.

**Comment by gsnedders** [comment]
Also, it seems some of the cario generated references differ depending on where they're generated. Argh.

**Comment by gsnedders** [comment]
The image differences are all in text tests, suspect this is down to fonts.

**Comment by gsnedders** [comment]
So yes, those that fail to create identical images are the exact set of tests that rely on Arial.

**Comment by gsnedders** [comment]
@sideshowbarker @jgraham @Ms2ger r?

**Pull request merged**
